File: templates/java/zh/text/code/invertLine.java
import stanford.卡雷尔.*;
public class 锥体S extends 卡雷尔 {
   /** 在这里开始执行 */
   public void run() {
      while(frontIsClear()) {
         反转锥体();
         move();
      }
      /* 防止fencepost bug */
      反转锥体();
   }

   /*  如果存在锥体则选择锥体否则为锥体 */
   public void 反转锥体() {
      /* 一个if / else语句 */
      if(beepersPresent()) {
         pickBeeper();
      } else {
         putBeeper();
      }
   }
}

File: templates/python/zh/text/code/invertLine.py
from 卡雷尔.stanford import *

＃程序的开始
def main():
   while front_is_clear(): 
      反转锥体()
      move()
   ＃防止击剑杆错误
   反转锥体()

＃如果锥体话则接一个锥体
＃否则放下锥体
def 反转锥体():
   ＃一条if / else语句
   if 锥体():
      选择锥体()
   else:
      放锥体()